modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -6181,13 +6184,30 @@ public String removeNonNameElementToCamelCase(String name) {
      * @return camelized string
      */
     protected String removeNonNameElementToCamelCase(final String name, final String nonNameElementPattern) {
-        String result = Arrays.stream(name.split(nonNameElementPattern))
-                .map(StringUtils::capitalize)
-                .collect(Collectors.joining(""));
-        if (result.length() > 0) {
-            result = result.substring(0, 1).toLowerCase(Locale.ROOT) + result.substring(1);
+        if (Boolean.parseBoolean(System.getProperty("openapi.generator.fix.camelize"))) {
+            // new behaviour with fix
+            String[] splitString = name.split(nonNameElementPattern);
```

Comment:
Can you change it to this so that it does not mutate the string in unintended ways? We just want to change casing, not strip whitespace.

String[] splitString = name.split(nonNameElementPattern, -1);